### ISSUE TITLE
Misc tarfile fixes

### ISSUE THREAD

**Issue by merwok** [python/cpython#58220]
BPO | [14012](https://bugs.python.org/issue14012)
--- | :---
Nosy | @gustaebel, @merwok, @bitdancer, @serhiy-storchaka
Files | <li>[tarfile-misc-bugs-3.2.diff](https://bugs.python.org/file24520/tarfile-misc-bugs-3.2.diff "Uploaded as text/plain at 2012-02-14.16:10:35 by @merwok")</li><li>[tarfile-misc-bugs-3.2-2.diff](https://bugs.python.org/file24601/tarfile-misc-bugs-3.2-2.diff "Uploaded as text/plain at 2012-02-22.14:26:02 by @gustaebel")</li><li>[tarfile-misc-bugs-3.2-3.diff](https://bugs.python.org/file24610/tarfile-misc-bugs-3.2-3.diff "Uploaded as text/plain at 2012-02-23.01:08:23 by @merwok")</li><li>[tarfile-misc-bugs-3.4.diff](https://bugs.python.org/file31146/tarfile-misc-bugs-3.4.diff "Uploaded as text/plain at 2013-08-04.08:16:36 by @serhiy-storchaka")</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2012-02-14.16:10:36.837>
labels = ['type-bug', 'library']
title = 'Misc tarfile fixes'
updated_at = <Date 2014-10-10.01:28:10.710>
user = 'https://github.com/merwok'
```

bugs.python.org fields:
```python
activity = <Date 2014-10-10.01:28:10.710>
actor = 'r.david.murray'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Library (Lib)']
creation = <Date 2012-02-14.16:10:36.837>
creator = 'eric.araujo'
dependencies = []
files = ['24520', '24601', '24610', '31146']
hgrepos = []
issue_num = 14012
keywords = ['patch']
message_count = 11.0
messages = ['153348', '153734', '153954', '154026', '181514', '186739', '186963', '194333', '195076', '195127', '228938']
nosy_count = 4.0
nosy_names = ['lars.gustaebel', 'eric.araujo', 'r.david.murray', 'serhiy.storchaka']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'needs patch'
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue14012'
versions = ['Python 2.7', 'Python 3.3', 'Python 3.4']
```

</p></details>

**Comment by iritkatriel** [comment]
The code has moved on since this patch was created. Can we close this?

**Comment by merwok** [comment]
FTR the first item was fixed in ce644a09ac73
